src/ResultWriter/faultoutput.f90

```diff
@@ -535,174 +541,179 @@ SUBROUTINE calc_FaultOutput( DynRup_output, DISC, EQN, MESH, MaterialVal, BND, t
-                iBndGP = DynRup_output%OutInt(iOutPoints,1)
-                ENDIF
-
-                OutVars = OutVars + 1
-                DynRup_output%OutVal(iOutPoints,1,OutVars)  = Vr
+            !calculation of the spatial derivatives of the rupture time
+            chi  = MESH%ELEM%BndGP_Tri(1,iBndGP)
+            tau  = MESH%ELEM%BndGP_Tri(2,iBndGP)
+            dt_dchi=0d0
+            dt_dtau=0d0
+            DO iDegFr = 1, nDegFr2d
+               call BaseGrad_Tri(phi2T,iDegFr,chi,tau,DISC%Galerkin%nPoly,DISC)
+               dt_dchi = dt_dchi + projected_RT(iDegFr) * phi2T(1)
+               dt_dtau = dt_dtau + projected_RT(iDegFr) * phi2T(2)
+            ENDDO
 
-              ENDIF
-              IF (DynRup_output%OutputMask(8).EQ.1) THEN
-                  OutVars = OutVars + 1
-                  DynRup_output%OutVal(iOutPoints,1,OutVars) = DISC%DynRup%output_Slip(iBndGP,iFace)
-              ENDIF
-              IF (DynRup_output%OutputMask(9).EQ.1) THEN
-                  OutVars = OutVars + 1
-                  DynRup_output%OutVal(iOutPoints,1,OutVars) = DISC%DynRup%output_PeakSR(iBndGP,iFace)
-              ENDIF
-              IF (DynRup_output%OutputMask(10).EQ.1) THEN
+            !For calculating the 2d Jacobian matrix we will need the nodal coordinates
+            IF (MESH%Fault%Face(iFace,1,1) == 0) THEN
+            ! iElem is in the neighbor domain
+            ! The neighbor element belongs to a different MPI domain
+            iNeighbor           = MESH%Fault%Face(iFace,1,2)          ! iNeighbor denotes "-" side
+            iLocalNeighborSide  = MESH%Fault%Face(iFace,2,2)
+            iObject  = MESH%ELEM%BoundaryToObject(iLocalNeighborSide,iNeighbor)
+            MPIIndex = MESH%ELEM%MPINumber(iLocalNeighborSide,iNeighbor)
+            !
+            xV(1:4) = BND%ObjMPI(iObject)%NeighborCoords(1,1:4,MPIIndex)
+            yV(1:4) = BND%ObjMPI(iObject)%NeighborCoords(2,1:4,MPIIndex)
+            zV(1:4) = BND%ObjMPI(iObject)%NeighborCoords(3,1:4,MPIIndex)
+            ELSE
+            !
+            ! get vertices
+            xV(1:4) = MESH%VRTX%xyNode(1,MESH%ELEM%Vertex(1:4,iElem))
+            yV(1:4) = MESH%VRTX%xyNode(2,MESH%ELEM%Vertex(1:4,iElem))
+            zV(1:4) = MESH%VRTX%xyNode(3,MESH%ELEM%Vertex(1:4,iElem))
+            ENDIF
+
+            SELECT CASE(iSide)
+             CASE(1)
+                xab = (/ xV(3)-xV(1), yV(3)-yV(1), zV(3)-zV(1) /)
+                xac = (/ xV(2)-xV(1), yV(2)-yV(1), zV(2)-zV(1) /)
+             CASE(2)
+                xab = (/ xV(2)-xV(1), yV(2)-yV(1), zV(2)-zV(1) /)
+                xac = (/ xV(4)-xV(1), yV(4)-yV(1), zV(4)-zV(1) /)
+             CASE(3)
+                xab = (/ xV(4)-xV(1), yV(4)-yV(1), zV(4)-zV(1) /)
+                xac = (/ xV(3)-xV(1), yV(3)-yV(1), zV(3)-zV(1) /)
+             CASE(4)
+                xab = (/ xV(3)-xV(2), yV(3)-yV(2), zV(3)-zV(2) /)
+                xac = (/ xV(4)-xV(2), yV(4)-yV(2), zV(4)-zV(2) /)
```

Comment:
Where do all these changes come from? Please don't "just" change the formatting and indentation.